src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -333,11 +352,19 @@ public WorkspacePanel(final MCreator mcreator) {
 			}
 		});
 
+		renameFolder.addActionListener(e -> {
+		IElement selected = list.getSelectedValue();
+		    if (selected instanceof FolderElement) {
+			    renameFolder();

```

Comment:
Pass selected here so you don't need to get it again inside method

Reply:
Fixed..... maybe?